### PR TITLE
Improvements to acquisition / CSV file loading.

### PR DESCRIPTION
**Pull request by melange396**
SQL batching improves ingestion speed significantly.
Loads files in name-sorted order for determinism and saner progress tracking.
Individual transactions for each file (instead of one transaction for the full collection of files) results in:
-an UPSERT failure means the entirety of the file will not be loaded (before a single line failure would preserve the preceding lines and attempt the following lines of the same file).
-lock should be released more frequently, meaning read access should not block as much while acquisition is in progress.

### CODE REVIEW

**Comment by melange396** [comment]
i started but then resisted the urge to rewrite more of this stuff in lieu of a smaller changelist...  there is some dead code that could be removed but i thought leaving it might provide context as to why the call structure is the way it is.  i also began to look into better handling of import failures, but decided that should go into its own issue, and i wasnt sure how representative my sample data files are.  can we try this out on the staging server?  if so, how can i do that?

**Comment by melange396** [comment]
some TODOs i didnt put in the code:
--do more speed testing with UPDATEs instead of clean INSERTs (UPDATE will be slower than INSERT in any case, but its unclear how much batching improves this)
--better deal with failure files - save error message somewhere, alert someone that errors happened, perhaps preserve a summary of what files were/werent loaded in each run, etc...
--ON DUPLICATE KEY - update (zero out) timestamp2 & direction ?? [ database.py:104 -ish ]
--make sure directories exist before writing files?  it might do this already?

**Comment by melange396** [comment]
im seeing 2-4x speed up for acquisition depending on which subsets of sample data i provide it with, though that is on a beefy but shared test machine.  @AlSaeed is there a particular tool used to get this profiling output, or was that just something you did by hand?  https://github.com/cmu-delphi/delphi-epidata/pull/133#issuecomment-645358029

**Comment by AlSaeed** [comment]
> im seeing 2-4x speed up for acquisition depending on which subsets of sample data i provide it with, though that is on a beefy but shared test machine. @AlSaeed is there a particular tool used to get this profiling output, or was that just something you did by hand? [#133 (comment)](https://github.com/cmu-delphi/delphi-epidata/pull/133#issuecomment-645358029)

This was done by hand via inserting boilerplate timing code around specific code blocks.

File: src/acquisition/covidcast/csv_to_database.py [hunk 1]
```diff
@@ -73,27 +73,21 @@ def archive_as_successful(path_src, filename, source):
...
-            row_values.geo_value,
-            row_values.value,
-            row_values.stderr,
-            row_values.sample_size)
-      except Exception as e:
+    csv_rows = csv_importer_impl.load_csv(path, geo_type)
+
+    if not csv_rows:
+      all_rows_valid = False
+      continue
```
Comment (krivard) [inline src/acquisition/covidcast/csv_to_database.py:81]: This continue will skip to the next file without archiving the busted one as failed

File: src/acquisition/covidcast/csv_to_database.py [hunk 2]
```diff
@@ -73,27 +73,21 @@ def archive_as_successful(path_src, filename, source):
...
-            geo_type,
-            time_value,
-            row_values.geo_value,
-            row_values.value,
-            row_values.stderr,
-            row_values.sample_size)
-      except Exception as e:
+    csv_rows = csv_importer_impl.load_csv(path, geo_type)
+
+    if not csv_rows:
```
Comment (krivard) [inline src/acquisition/covidcast/csv_to_database.py:79]: `csv_rows` is a generator and will always be truthy; you probably want `any()` or `all()` here instead

File: src/acquisition/covidcast/database.py [hunk 3]
```diff
@@ -251,41 +339,6 @@ def update_timeseries_timestamp2(
     args = (source, signal, time_type, geo_type, geo_value)
     self._cursor.execute(sql, args)
 
-  def get_covidcast_meta(self):
```
Comment (krivard) [inline src/acquisition/covidcast/database.py:254]: Why was this removed?

File: src/acquisition/covidcast/csv_to_database.py (hunk 1 above)
Comment (melange396) [reply src/acquisition/covidcast/csv_to_database.py:81]: good catch

File: src/acquisition/covidcast/csv_to_database.py (hunk 2 above)
Comment (melange396) [reply src/acquisition/covidcast/csv_to_database.py:79]: good catch

File: src/acquisition/covidcast/database.py (hunk 3 above)
Comment (melange396) [reply src/acquisition/covidcast/database.py:254]: whoops!  looks like it was from a merge conflict...  i dont know how i missed that
